common/workunit/workunit.cpp

```diff
@@ -6863,40 +6834,51 @@ const char *clusterTypeString(ClusterType clusterType, bool lcrSensitive)
     throwUnexpected();
 }
 
-bool extractFromWorkunitDAToken(const char * distributedAccessToken, StringBuffer * wuid, StringBuffer * user, StringBuffer * privKey)
+//Does given token appear to be in the right format. KeyFile and signature are optional
+//    HPCC[u=user,w=wuid]keyFile;signature
+bool isWorkunitDAToken(const char * distributedAccessToken)
 {
-    if (strncmp(distributedAccessToken, "HPCC[", 5))
+    if (strncmp(distributedAccessToken, "HPCC[", 5) ||
+        !strstr(distributedAccessToken+5, "u=") ||
+        !strchr(distributedAccessToken+5, ',')  ||
+        !strstr(distributedAccessToken+5, "w=") ||
+        !strchr(distributedAccessToken+5, ']')  ||
+        !strchr(distributedAccessToken+5, ';'))
     {
-        DBGLOG("Not a valid workunit distributed access token");
         return false;
     }
+    return true;//appears to be a workunit token
+}
 
-    const char * finger;
-    if (user)
+bool extractFromWorkunitDAToken(const char * distributedAccessToken, StringBuffer * wuid, StringBuffer * user, StringBuffer * privKey)
+{
+    if (!isWorkunitDAToken(distributedAccessToken))
     {
-        finger = strstr(distributedAccessToken+5, "u=");
-        if (finger)
-        {
-            finger += 2;
-            while (*finger && *finger != ',' && *finger != ']')
-                user->append(1, (const char *)finger++);
-        }
+        DBGLOG("Not a valid workunit distributed access token");
+        return false;
     }
 
-    if (wuid)
+    //Isolate string between []
+    StringBuffer tokenValues(distributedAccessToken + 5);
+    tokenValues.replace(']', (char)0);
```

Comment:
This works, but seems a bit messy to me.  Rather then embedding a 0 in the string I would find the ] first and then create tokenValues string to contain the exact content already.  I would also save off the position of the ] for use below.